ProjFS.Mac/PrjFSKext/kernel-header-wrappers/vnode.h

```diff
@@ -1,7 +1,21 @@
 #pragma once
 
+#if defined(KEXT_UNIT_TESTING) && !defined(TESTABLE_KEXT_TARGET) // Building unit tests
+
+#include <time.h> // vnode.h below references struct timespec
+// Definitions to make the kernel vnode.h compile in user space
+typedef struct ipc_port* ipc_port_t;
+enum uio_rw {};
+enum uio_seg {};
```

Comment:
@jrbriggs the lines is 'magic' required to use the Kernel/sys/vnode.h header.  If we hadn't done this we'd be looking at redefining kext objects.  @pmj feel free to elaborate more.

Reply:
We're essentially cheating by pulling the kernel versions of *just* `sys/vnode.h` and `sys/mount.h`. There are cut-down user space versions of those files but they don't contain any prototypes for the in-kernel functions of course. (`vnode_getattr()` and such) So by pulling in the kernel versions of those headers we avoid having to duplicate the prototypes, which saves us some work but has an extra advantage in that we can't make mistakes in the signature of the mocked versions of those functions without the compiler yelling at us.

Of course, those kernel headers have dependencies for various types etc. themselves, but those aren't all met by the userspace versions of the headers they include. So we have to fake them sufficiently to get the compiler to stop complaining. This isn't a problem as we don't actually use any of the APIs that use these specific types.